internal/api/core/core_test.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+import (
+	"encoding/hex"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"path/filepath"
+	"testing"
+
+	"github.com/gavv/httpexpect/v2"
+	"github.com/perses/perses/internal/api/utils"
+	"github.com/perses/perses/internal/test"
+	apiConfig "github.com/perses/perses/pkg/model/api/config"
+	"github.com/perses/perses/pkg/model/api/v1/secret"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/assert"
+)
+
+// We cannot reuse in this package what is inside internal/api/e2e because it would create a
```

Comment:
It would be very disappointing :/ If it's an e2e test, it should be located in e2e folder

Reply:
Well e2e is properly badly named but this is definetely the kind of tests that we do in that folder. What we do in the WithServer of the "e2eframework" we've built is not very different from what you are doing in this withServer function. Considering the way we architectured, the place of that test is in a cors_test.go in e2e /api folder.

Reply:
Sounds good, I moved and renamed the file.